lib/python/rosie/ws.py

```diff
@@ -61,35 +77,90 @@ def __init__(self, *args, **kwargs):
                 ResourceLocator.default().get_util_home(
                     "lib", "html", "template", "rosie-disco")))
         db_url_map = {}
-        for key, node in rose_conf.get(["rosie-db"]).value.items():
+        for key, node in list(rose_conf.get(["rosie-db"]).value.items()):
             if key.startswith("db.") and key[3:]:
                 db_url_map[key[3:]] = node.value
         self.db_url_map = db_url_map
         if not self.db_url_map:
             self.db_url_map = {}
-        for key, db_url in self.db_url_map.items():
-            setattr(self, key, RosieDiscoService(self.props, key, db_url))
 
-    @cherrypy.expose
-    def index(self, *_):
+        # Set-up the Tornado application request-handling structure.
+        service_root = r"/" + RosieDiscoServiceApplication.NS + r"/?"
```

Comment:
Is there a reason why you've called `RosieDiscoServiceApplication.NS` here and `self.NS` at ~line 101?
What does NS stand for? If I need to ask should it be more self-explanatory?

Reply:
> Is there a reason why you've called RosieDiscoServiceApplication.NS here and self.NS at ~line 101?

There is not, no, other than that I had not noticed that both variants had been used. Good observation! I will make these consistent & since they are referring to a string i.e. a constant I think ``self.`` would be most appropriate to handle inheritance safely.

> What does NS stand for? If it's a standard tornado idiom?

I'm not actually sure! It is not related to Tornado, but to web services, or at least as coded in our codebases. I suspect it stands for 'namespace', as it has the root in a chain ``NS``, ``UTIL`` (utility, surely), ``HOST`` & then ``PORT``, as in:

https://github.com/metomi/rose/blob/e88db4259c5a31d578bc86615b2cd8b7b98ba556/lib/python/rose/ws.py#L36

where ``NS`` cases relevant to us are ``rose`` & ``cylc``.

> If I need to ask should it be more self-explanatory?

Yes, I endorse a clearer name, though quite how to shorten 'namespace' clearly I am not sure. Perhaps the clearest choice is not to shorten it.

Reply:
do you need to shorten "namespace"? If you don't I'd suggest just commenting it.

Reply:
Agreed, let's go with ``NAMESPACE``. No ambiguity there, though we will have to presume it is the correct term given no-one we have asked at this stage is certain!